sovryn-token-bridge/federator/src/main.js

```diff
@@ -71,21 +74,47 @@ const mainFederator = new Federator(
     config,
     log4js.getLogger('MAIN-FEDERATOR'),
     web3,
-    chatBot,
+    p2pNode,
+    chatBot
 );
 
 const sideFederator = new Federator(
     {
         ...config,
         mainchain: config.sidechain,
         sidechain: config.mainchain,
-        storagePath: `${config.storagePath}/side-fed`
+        storagePath: `${config.storagePath}/side-fed`,
     },
     log4js.getLogger('SIDE-FEDERATOR'),
     web3,
-    chatBot,
+    p2pNode,
+    chatBot
 );
 
+p2pNode.net.onMessage(async (msg) => {

```

Comment:
I think that this should be a part of the Federator class, though it also requires some restructuring in the way federator.run() is called.